cmd/doctor.go

```diff
@@ -42,13 +70,23 @@ var checklist = []check{
 	{
 		// NOTE: this check should be the first in the list
 		title:            "Check paths and basic configuration",
+		name:             "paths",
+		isDefault:        true,
 		f:                runDoctorPathInfo,
 		abortIfFailed:    true,
 		skipDatabaseInit: true,
 	},
 	{
-		title: "Check if OpenSSH authorized_keys file id correct",
-		f:     runDoctorLocationMoved,
+		title:     "Check if OpenSSH authorized_keys file id correct",
+		name:      "authorized_keys",
+		isDefault: true,
+		f:         runDoctorLocationMoved,
+	},
+	{
```

Comment:
Before the first check that operates with the database, we should add two barrier checks as well:

- Check that no instance is running; I've seen the discussion in Discord, I know it's not trivial but a simplistic check will be better than no check.
- Check that the database is running the current version (from `migration.go`). Otherwise we risk either altering models with code from a different version, or producing an unintended upgrade. Or both, if we're unadvertently downgrading.

The version check also works as a "can connect" check.

Reply:
I'm not actually certain that we have to have gitea not running - even on sqlite since we have unlock notify I think we can edit in two processes at the same time.

Reply:
In any case that should probably be another PR.

Reply:
OK, with our current fixes there will be no harm, but IFAIK there's a xorm cache that's instance-based.

The version check, however, is important IMHO.

Reply:
Done. I'll update this PR to be a general improvement of doctor then.